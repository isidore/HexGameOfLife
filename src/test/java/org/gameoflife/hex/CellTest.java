package org.gameoflife.hex;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class CellTest {
    @Test
    void oddsGoTogether() {
        assertEquals(true, Cell.isValid(1,1));
        assertEquals(true, Cell.isValid(2,2));
        assertEquals(false, Cell.isValid(2,1));
        assertEquals(false, Cell.isValid(1,2));
    }

    @Test
    void getLevelOneNeighbours() {
        GameOfLifeBoard gameOfLifeBoard = new GameOfLifeBoard(Cell.getLevelOneNeighbours(2, 4));
        Approvals.verify(gameOfLifeBoard);
    }
    @Test
    void getLevelTwoNeighbours() {
        GameOfLifeBoard gameOfLifeBoard = new GameOfLifeBoard(Cell.getLevelTwoNeighbours(6,4));
        gameOfLifeBoard.setAlive(6,4);
        Approvals.verify(gameOfLifeBoard);
    }

}
