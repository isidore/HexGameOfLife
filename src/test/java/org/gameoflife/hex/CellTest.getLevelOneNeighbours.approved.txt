_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  X   X   _   _   _ 
X   _   X   _   _   
  X   X   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
